hplot/label.go

```diff
@@ -0,0 +1,107 @@
+// Copyright ©2020 The go-hep Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package hplot
+
+import (
+	"image/color"
+	"math"
+
+	"gonum.org/v1/plot"
+	"gonum.org/v1/plot/plotter"
+	"gonum.org/v1/plot/vg"
+	"gonum.org/v1/plot/vg/draw"
+)
+
+type Label struct {
+	Text      string         // Text of the label
+	X, Y      float64        // Position of the label
+	TextStyle draw.TextStyle // Text style of the label
+
+	// Normalized indicates whether the label position
+	//is in data coordinates  or normalized with regard
+	// to the canvas space.
+	// When normalized, the label position is assumed
+	// to fall in the [0, 1] interval.
+	Normalized bool
+}
+
+// Plot implements the Plotter interface,
+// drawing the label on the canvas.
+func (lbl Label) Plot(c draw.Canvas, p *plot.Plot) {
+	lbls := lbl.labels(c, p)
+	lbls.Plot(c, p)
+}
+
+// DataRange returns the minimum and maximum x and
+// y values, implementing the plot.DataRanger interface.
+func (lbl Label) DataRange() (xmin, xmax, ymin, ymax float64) {
+
+	if lbl.Normalized {
+		return math.Inf(+1), math.Inf(-1), math.Inf(+1), math.Inf(-1)
+	}
+
+	pLabels := lbl.labels(draw.Canvas{}, nil)
+	return pLabels.DataRange()
+}
+
+// GlyphBoxes returns a GlyphBoxe, corresponding
+// to the label, implementing the plot.GlyphBoxer interface.
+func (lbl Label) GlyphBoxes(p *plot.Plot) []plot.GlyphBox {
+
+	if lbl.Normalized {
+		// FIXME[rmadar]: this crashes and I don't understand why.
```

Comment:
probably because `lbl.TextStyle` is the zero value of `draw.TextStyle`? (and thus the call to `lbl.TextStyle.Rectangle` is using zero values of a few of its fields)

either:
- introduce a little helper method `func (lbl *Label) style() draw.TextStyle` (and use it throughout)
- re-introduce your `NewLabel(...) *Label` function (in which we make sure to create a `TextStyle` field value with a meaningful default value) (and make `*Label` implement `plot.Plotter`, etc...)

perhaps option 2 above is better afterall (apologies for the back and forth on this matter)